samples/Gonda/Gonda.Client/Program.cs

```diff
@@ -0,0 +1,158 @@
+            
+            await CreateUser();
+            var tokens = await GetTokens();
+            
+            var authenticatedClient = CreateAuthenticatedClient(tokens.AccessToken);
+
+            await SayHello(authenticatedClient);
+            await SaySecret(authenticatedClient);
+            
+            Console.WriteLine("Press any key to exit...");
+            Console.ReadKey();
+        }
+
+        private static Greeter.GreeterClient CreateAnonymousClient()
+        {
+            var channel = GrpcChannel.ForAddress(ApiUrl);
+            
+            return new Greeter.GreeterClient(channel);
+        }
+
+        private static Greeter.GreeterClient CreateAuthenticatedClient(string token)
+        {
+            var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            
+            var opt = new GrpcChannelOptions()
+            {
+                HttpClient = httpClient, 
+            };
+            
+            var channel = GrpcChannel.ForAddress(ApiUrl, opt);
+            
+            return new Greeter.GreeterClient(channel);
+        }
+
+        private static async Task SayHello(Greeter.GreeterClient client)
+        {
+            try
+            {
+                Console.WriteLine("Attempting to say hello to gRPC service...");
+
+                var reply = await client.SayHelloAsync(new HelloRequest {Name = "GreeterClient"});
+
+                Console.WriteLine("Response: " + reply.Message);
+            }
+            catch (RpcException ex)
+            {
+                Console.WriteLine("Response: " + ex.Status);
+            }
+            
+            Console.WriteLine();
+        }
+        
+        private static async Task SaySecret(Greeter.GreeterClient client)
+        {
+            try
+            {
+                Console.WriteLine("Attempting to tell a secret to gRPC service...");
+
+                var reply = await client.SaySecretAsync(new SecretRequest {Secret = "Some thing"});
+
+                Console.WriteLine("Response: " + reply.Message);
+            }
+            catch (RpcException ex)
+            {
+                Console.WriteLine("Response: " + ex.Status);
+            }
+            
+            Console.WriteLine();
+        }
+        
+        private static async Task CreateUser()
+        {
+            Console.WriteLine("Creating user...");
+            
+            var authClient = new HttpClient()
+            {
+                BaseAddress = new Uri(AuthUrl)
+            };
+
+            var request = new
+            {
+                Email = "test@email.com",
+                Password = "Gonda~1",
+                ConfirmPassword = "Gonda~1"
+            };
+            
+            var content = JsonConvert.SerializeObject(request);
+
+            var response = await authClient.PostAsync("Account/register", 
```

Comment:
You may want to use the new HTTP JSON APIs (w/ `JsonContent`) instead of using JSON.NET + `StringContent`.